w_common/lib/src/common/disposable.dart

```diff
@@ -534,12 +534,7 @@ class Disposable implements _Disposable, DisposableManagerV7, LeakFlagger {
     });
     _internalDisposables.add(disposable);
 
-    completer.future.catchError((e) {
-      if (!_isDisposedOrDisposing) {
-        _logUnmanageMessage(completer);
-        _internalDisposables.remove(disposable);
-      }
-    }).then((_) {
+    completer.future.then((value) => null).catchError((_) {}).whenComplete(() {
```

Comment:
`.catchError((_) {})` is a bit obscure and nuanced, thoughts on adding a quick comment here explaining why its necessary?